src/items/tools/tool.js

```diff
@@ -35,25 +35,35 @@ export default class Tool extends Item {
         console.log("Combining", this.name, item.name);
 
         // Craft item if possible
-        if (recipe.className !== "" && item.numItems >= recipe.numRequiredItems) {
+        if (item.numItems >= recipe.numRequiredItems) {
             const dashboard = characterData.getScene(CONSTANTS.SCENES.DASHBOARD);
-            let newItem = await getItemClass(recipe.className, dashboard);
 
-            // Get name before adding it to inventory because
-            // if it's a duplicate it will be destroyed
-            const newItemName = newItem.name;
+            if (recipe.className !== "") {
+                let newItem = await getItemClass(recipe.className, dashboard);
 
-            // Item was added
-            if (dashboard.inventory.addToInventory(newItem)) {
-                item.setNumItems(item.numItems - recipe.numRequiredItems);
+                // Get name before adding it to inventory because
+                // if it's a duplicate it will be destroyed
+                const newItemName = newItem.name;
+
+                // Item was added
+                if (dashboard.inventory.addToInventory(newItem)) {
+                    item.setNumItems(item.numItems - recipe.numRequiredItems);
 
+                    // TODO: differentiate between skills when we support more
+                    outputString =
+                        "Fletched " + newItemName + ".";
+                    characterData.addSkillXp({ fletching: recipe.xpGiven });
+                }
+            } else {
+                // No new item is created
+                item.setNumItems(item.numItems - recipe.numRequiredItems);
                 outputString =
-                    "Crafted " + newItemName + ", added " + recipe.xpGiven + "xp";
-                characterData.addSkillXp({fletching: recipe.xpGiven});
+                    "Firemaked " + item.name + ".";
```

Comment:
Since this method is so different from the fletching one I think we should refactor this a bit. A couple of options:

1. Override `craft` method in child class
2. Require child to pass in action callback function into the craft method to handle custom logic
3. Create sub classes under tools to handle things in common, like grouping tools that create items as one subclass

Reply:
Went with 1